Projects/UOContent/Mobiles/Animals/Misc/Sheep.cs

```diff
@@ -84,32 +81,7 @@ public void Carve(Mobile from, Item item)
         public override void OnThink()
         {
             base.OnThink();
-            Body = Core.Now >= m_NextWoolTime ? 0xCF : 0xDF;
-        }
-
-        public override void Serialize(IGenericWriter writer)
-        {
-            base.Serialize(writer);
-
-            writer.Write(1);
-
-            writer.WriteDeltaTime(m_NextWoolTime);
-        }
-
-        public override void Deserialize(IGenericReader reader)
-        {
-            base.Deserialize(reader);
-
-            var version = reader.ReadInt();
-
-            switch (version)
-            {
-                case 1:
-                    {
-                        NextWoolTime = reader.ReadDeltaTime();
-                        break;
-                    }
-            }
+            Body = Core.Now >= _nextWoolTime ? 0xCF : 0xDF;
```

Comment:
This line:
```cs
Body = Core.Now >= _nextWoolTime ? 0xCF : 0xDF;
```

Shows up several times. Let's put it into a function and reuse it.
That method should have the `[MethodImpl(MethodImplOptions.AggressiveInlining)]`

Also we need an `AfterDeserialization` that calls this logic too.

Reply:
like this?

```cs
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public void SheepBody() => Body = Core.Now >= _nextWoolTime ? 0xCF : 0xDF;
```

Reply:
Yes, this looks good! Lets name it `UpdateSheepBody()`